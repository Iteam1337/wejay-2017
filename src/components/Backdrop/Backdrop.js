import './Backdrop.css'
import React from 'react'

const Backdrop = ({ track }) => {
  if (!track) {
    return null
  }

  return (
    <div
      className="Backdrop"
      style={{ backgroundImage: `url(${track.album.images[0].url})` }}
    />
  )
}

export default Backdrop
