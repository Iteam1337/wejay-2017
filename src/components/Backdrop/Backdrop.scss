.Backdrop {
  background-size: cover;
  background-position: center center;
  bottom: 0;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
}
