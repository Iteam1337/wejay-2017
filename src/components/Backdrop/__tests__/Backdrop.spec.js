import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Backdrop from '../Backdrop'

describe('components/Backdrop', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <Backdrop
        track={{
          album: {
            images: [
              {
                url: 'test.png'
              }
            ]
          }
        }}
      />
    )
  })

  it('returns null', () => {
    component.setProps({ track: undefined })

    expect(component.type()).toBeNull()
  })

  it('returns Backdrop', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
