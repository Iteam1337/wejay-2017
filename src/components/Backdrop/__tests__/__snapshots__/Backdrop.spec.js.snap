// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Backdrop returns Backdrop 1`] = `
<div
  className="Backdrop"
  style={
    Object {
      "backgroundImage": "url(test.png)",
    }
  }
/>
`;
